atc/worker/worker.go

```diff
@@ -609,9 +611,13 @@ func (worker *gardenWorker) cloneRemoteVolumes(
 			"dest-worker": inputVolume.WorkerName(),
 		}
 
+
+		maxInFlight <- 1
+
 		g.Go(func() error {
```

Comment:
while, this PR didn't result in the change, is there a reason why the returned `error` isn't used ?

Reply:
g.Go() manages the errors returned by all the go routines. When one of the goroutines returns an error. It is return by g.Wait() as seen in line 635. We use the error there.

Reply:
nvm, I just saw the impl of `g.Go` & `g.Wait`

Reply:
an errgroup is basically a wait group that waits on errors instead of .Done()s to be called on a sync.waitgroup.